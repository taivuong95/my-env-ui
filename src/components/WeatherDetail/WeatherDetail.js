import React from 'react';
import './WeatherDetail.scss'

const WeatherDetail = () => {
    return (
        <div className='weather-detail'>
          <div className='weather-detail__block'>
            <div className='weather-detail__icon--cloud'></div>
            <div className='weather-detail__info'>
              <div className='weather-detail__info--weather'>Cloudy</div>
              <div className='weather-detail__info--status'>
                <span className='weather-detail__info--temp-status'>
                  29.2 C
                </span>
                <span className='weather-detail__info--humid-status'>73%</span>
              </div>
            </div>
          </div>
        </div>
    )
}

export default WeatherDetail