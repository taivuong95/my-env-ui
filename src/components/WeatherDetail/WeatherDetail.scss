.weather-detail{
    display: flex;
      justify-content: center;
      align-items: center;
      height: 40%;
  }
  
  .weather-detail__block{
    display: flex;
    justify-content: center;
  }
  
  .weather-detail__icon--cloud{
    width: 8rem;
    height: 8rem;
    background-size: contain;
    background-image: url(../../assets/img/cloud.svg);
  }
  
  .weather-detail__info{
    margin-left: 3rem;
    display: flex;
    flex: 1 1;
    flex-direction: column;
    justify-content: space-between;
    height: 8rem;
    width: 18rem;
  }
  
  .weather-detail__info--weather{
    font-size: 3rem;
  }
  .weather-detail__info--status{
    display: flex;
    justify-content: space-between;
    font-size: 2rem;
  }
  
  .weather-detail__info--temp-status::before{
    display: inline-block;
    content: "";
    background-image: url(../../assets/img/temp.svg);
    background-size: contain;
    width: 2rem;
    height: 2rem;
  }
  .weather-detail__info--humid-status::before{
    display: inline-block;
    content: "";
    background-image: url(../../assets/img/humid.svg);
    background-size: contain;
    width: 2rem;
    height: 2rem;
  }