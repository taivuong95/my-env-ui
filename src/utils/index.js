export const KelvinToCelsius = (kelvin) => {
    return kelvin - 273.15;
}