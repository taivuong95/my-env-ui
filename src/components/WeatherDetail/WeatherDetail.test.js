import React from 'react';
import WeatherDetail from './WeatherDetail';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { findByTestAttr } from '../../utils/index';
import Context from '../../contexts/Context';
import Loader from '../UI/Loader';

const value = {
    getWeatherInfo: jest.fn(),
    loading: true
}

describe('WeatherDetail Component', () => {
    test("expect to render WeatherDetail component", () => {
        const wrapper = mount(<Context.Provider value={value}><WeatherDetail /> </Context.Provider>);
        expect(toJson(wrapper)).toMatchSnapshot();
    })
    test('renders without error', () => {
        const wrapper = mount(<Context.Provider value={value}><WeatherDetail /> </Context.Provider>)
        const weatherDetailComponent = findByTestAttr(wrapper, "weather-detail");
        expect(weatherDetailComponent.length).toBe(0);
    });
    test(" render a loading before api call success(loading is true)", () => {
        const wrapper = mount(<Context.Provider value={value}><Loader /> </Context.Provider>)
        expect(wrapper.find(".loader").exists()).toBeTruthy();
    });

})