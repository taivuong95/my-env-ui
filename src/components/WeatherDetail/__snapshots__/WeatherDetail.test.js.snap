// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherDetail Component expect to render WeatherDetail component 1`] = `
Array [
  <WeatherDetail>
    <Loader>
      <div
        className="loader"
      />
    </Loader>
  </WeatherDetail>,
  " ",
]
`;
